llvm/lib/ObjCopy/ELF/ELFObjcopy.cpp

```diff
@@ -298,6 +298,10 @@ static Error updateAndRemoveSymbols(const CommonConfig &Config,
          Config.SymbolsToLocalize.matches(Sym.Name)))
       Sym.Binding = STB_LOCAL;
 
+    for (auto &[Matcher, VisibilityType] : ELFConfig.SymbolsToSetVisibility)
+      if (Matcher.matches(Sym.Name) && Sym.getShndx() != SHN_UNDEF)
```

Comment:
Why is there a special case for undefined symbol? A protected/hidden undefined should be allowed.

The following code generates a hidden undefined symbol.

```
#pragma GCC visibility push(hidden)
void bar();
void foo() {
  bar();
}
```